### PR TITLE
Setup docker

### PR DESCRIPTION
**Pull request by virsavik**
- Make a docker-compose file to run 3 service, such as: app, postgres, migrate.
- Make a Dockerfile to build image for app.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+FROM golang:1.18.3-alpine3.16
+
+WORKDIR /opt
+
+COPY --from=0 /app .
+
+EXPOSE 8000
+
+CMD [ "/opt/app" ]
```
Comment (vinhnv1) [inline Dockerfile:19]: Add new blank line to resolve the warning of red circle

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,7 @@
+include .env
+export
+
+.PHONY: migrate-up
+migrate-up:
```
Comment (vinhnv1) [inline Makefile:5]: How to start db and app this make file

File: data/migrations/001_setup_users.up.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    "email" TEXT NOT NULL DEFAULT '',
+    "password" TEXT NOT NULL DEFAULT '',
+    "phone" TEXT NOT NULL DEFAULT '',
+    "role" TEXT NOT NULL DEFAULT '', -- ADMIN, GUEST
+    "is_active" BOOL NOT NULL DEFAULT true,
+    "created_at" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
+    "updated_at" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW()
+);
+
+COMMIT;
```
Comment (vinhnv1) [inline data/migrations/001_setup_users.up.sql:15]: Add new blank line below

Comment (virsavik) [reply data/migrations/001_setup_users.up.sql:15]: Ok anh

**Pull request merged**